### PR TITLE
Spsh 929

### PR DESCRIPTION
**Pull request by clauyan**
# Description
Extended the login-info function in the auth-controller, so that it queries keycloak for the last password change. Keycloak only stores the creation-timestamp for a password, which is overwritten on every subsequent change to the password. The logic to determine whether the password has been changed or it is the initial password lives in the `KeycloakUserService`. The data is only sent if the password was actually changed after account creation.

## Links to Tickets or other pull requests
[Ticket](https://ticketsystem.dbildungscloud.de/browse/SPSH-925)
[FE PR](https://github.com/dBildungsplattform/schulportal-client/pull/236)

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Review by marode-cap — APPROVED** [review]
Approved, but check the two comments

File: src/modules/authentication/api/userinfo.response.ts [hunk 1]
```diff
@@ -75,5 +86,6 @@ export class UserinfoResponse {
         this.birthdate = info.personFields.geburtsdatum?.toISOString();
         this.updated_at = info.personFields.updatedAt.getTime() / 1000;
         this.personenkontexte = personenkontexte;
+        this.password_updated_at = extension?.password_updated_at;
```
[the remark this answers is not in the document]
Comment (clauyan) [reply src/modules/authentication/api/userinfo.response.ts:89]: I am not aware of a reason other than that `password_updated_at` (and presumably `updated_at`) is straight from keycloak and that is the format the keycloak-devs use. Is there already a global convention for timestamps in place? I'm open to add the conversion here, but I think this should be done consistently throughout the codebase.

File: src/modules/authentication/api/userinfo.response.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { DBiamPersonenkontextRepo } from '../../personenkontext/persistence/dbiam-personenkontext.repo.js';
+import { OrganisationRepository } from '../../organisation/persistence/organisation.repository.js';
+import { RolleRepo } from '../../rolle/repo/rolle.repo.js';
+import { PersonenkontextRolleFieldsResponse } from './personen-kontext-rolle-fields.response.js';
+
+describe('UserinfoResponse', () => {
+    const permissions: PersonPermissions = new PersonPermissions(
+        jest.fn() as unknown as DBiamPersonenkontextRepo,
+        jest.fn() as unknown as OrganisationRepository,
+        jest.fn() as unknown as RolleRepo,
```
[the remark this answers is not in the document]
Comment (clauyan) [reply src/modules/authentication/api/userinfo.response.spec.ts]: Addressed

File: src/modules/authentication/api/userinfo.response.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marode-cap) [reply src/modules/authentication/api/userinfo.response.ts:89]: Most of our timestamps are already JS Date-objects. If keycloak returns them as a number it's fine by me, but converting them to Date-Objects would be more idiomatic in my opinion

[the remark this answers is not in the document]
Comment (clauyan) [reply src/modules/authentication/api/userinfo.response.ts]: I agree and changed this to use Date instead.

**Pull request merged**